pkg/controllers/v1alpha1/databackup/databackup_controller.go

```diff
@@ -58,15 +59,30 @@ func NewDataBackupReconciler(client client.Client,
 	return r
 }
 
+func (r *DataBackupReconciler) Build(object client.Object) (dataoperation.OperationInterface, error) {
+	dataBackup, ok := object.(*datav1alpha1.DataBackup)
+	if !ok {
+		return nil, fmt.Errorf("object %v is not a DataBackup", object)
+	}
+
+	return &dataBackupOperation{
+		Client:     r.Client,
+		Log:        r.Log,
+		Recorder:   r.Recorder,
+		dataBackup: dataBackup,
+	}, nil
+}
+
```

Comment:
Do we have to implement the `Build` function for each kind of data operation? What about something like:
```
func Build(object client.Object) (dataoperation.OperationInterface, error) {
   // 1. type switch for 4 kind of data operations
  // 2. according to the data operation type, return dataBackupOpeartion/dataLoadOperation/dataProcessOperation/dataMigrateOperation
}
```

Reply:
> Do we have to implement the `Build` function for each kind of data operation? What about something like:
> 
> ```
> func Build(object client.Object) (dataoperation.OperationInterface, error) {
>    // 1. type switch for 4 kind of data operations
>   // 2. according to the data operation type, return dataBackupOpeartion/dataLoadOperation/dataProcessOperation/dataMigrateOperation
> }
> ```

I think the benefit is that if there is a new operation introduced,  the build function should be implemented or it will fail to build.